test/e2e/workflow/workflow.test.ts

```diff
@@ -0,0 +1,380 @@
+        expect(state?.workflowFailureDetails).toBeUndefined();
+        expect(state?.runtimeStatus).toEqual(WorkflowRuntimeStatus.COMPLETED);
+        expect(state?.createdAt).toBeDefined();
+        expect(state?.lastUpdatedAt).toBeDefined();
+        expect(expectedCompletionSecond).toBeLessThanOrEqual(actualCompletionSecond!);
+    }, 31000);
+
+    it("should wait for external events with a timeout - true", async () => {
+        const shouldRaiseEvent = true;
+        const workflow: TWorkflow = async function* (ctx: WorkflowContext, _: any): any {
+            const approval = ctx.waitForExternalEvent("Approval");
+            const timeout = ctx.createTimer(3);
+            const winner = yield ctx.whenAny([approval, timeout]);
+
+            if (winner == approval) {
+                return "approved";
+            } else {
+                return "timed out";
+            }
+        };
+
+        workflowRuntime.registerWorkflow(workflow);
+        await workflowRuntime.start();
+
+        // Send events to the client immediately
+        const id = await workflowClient.scheduleNewWorkflow(workflow);
+
+        if (shouldRaiseEvent) {
+            workflowClient.raiseEvent(id, "Approval");
+        }
+
+        const state = await workflowClient.waitForWorkflowCompletion(id, undefined, 30);
+
+        expect(state);
+        expect(state?.runtimeStatus).toEqual(WorkflowRuntimeStatus.COMPLETED);
+
+        if (shouldRaiseEvent) {
+            expect(state?.serializedOutput).toEqual(JSON.stringify("approved"));
+        } else {
+            expect(state?.serializedOutput).toEqual(JSON.stringify("timed out"));
+        }
+    }, 31000);
+
+    it("should wait for external events with a timeout - false", async () => {
+        const shouldRaiseEvent = false;
+        const workflow: TWorkflow = async function* (ctx: WorkflowContext, _: any): any {
+            const approval = ctx.waitForExternalEvent("Approval");
+            const timeout = ctx.createTimer(3);
+            const winner = yield ctx.whenAny([approval, timeout]);
+
+            if (winner == approval) {
+                return "approved";
+            } else {
+                return "timed out";
+            }
+        };
+
+        workflowRuntime.registerWorkflow(workflow);
+        await workflowRuntime.start();
+
+        // Send events to the client immediately
+        const id = await workflowClient.scheduleNewWorkflow(workflow);
+
+        if (shouldRaiseEvent) {
+            workflowClient.raiseEvent(id, "Approval");
+        }
+
+        const state = await workflowClient.waitForWorkflowCompletion(id, undefined, 30);
+
+        expect(state);
+        expect(state?.runtimeStatus).toEqual(WorkflowRuntimeStatus.COMPLETED);
+
+        if (shouldRaiseEvent) {
+            expect(state?.serializedOutput).toEqual(JSON.stringify("approved"));
+        } else {
+            expect(state?.serializedOutput).toEqual(JSON.stringify("timed out"));
+        }
+    }, 31000);
```

Comment:
Can we combine these tests?

Reply:
Actually, I tried to combine them, but they interfered with each other, so I separated them to utilize the `beforeEach` method defined in the test suit.